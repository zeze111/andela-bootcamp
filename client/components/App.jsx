import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Homepage from './homepage/Homepage'
import SignupPage from './signup/Signuppage'
import NavigationBar from './NavigationBar';
import Footer from './Footer';
import FlashMessageList from './flash/FlashMessageList';
import AddRecipePage from './addRecipe/AddRecipePage';
import Profile from './user/Profile';
import Favorites from './user/Favorites';
import '../assets/init';


class App extends Component {
  render() {
    return (
      <Router>
        <div id='wrap'>
          <NavigationBar />
          <FlashMessageList />
          <Switch>
            <Route path='/' exact component={Homepage} />
            <Route path='/signup' component={SignupPage} />
            <Route path='/addRecipe' component={AddRecipePage} />
            <Route path='/user' component={Profile} />
          </Switch>
          <Footer />
        </div>
      </Router>
    );
  }
}
export default App;
