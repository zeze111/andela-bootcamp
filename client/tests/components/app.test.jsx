import React from 'react';

import App from '../../components/App';

describe('Test for routes', () => {
  it('run all the routes', () => {
    const wrapper = shallow(<App /> );
    expect(wrapper).toMatchSnapshot()
  })
})
