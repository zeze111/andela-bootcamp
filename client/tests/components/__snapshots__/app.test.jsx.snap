// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for routes run all the routes 1`] = `
<BrowserRouter>
  <div
    id="wrap"
  >
    <Connect(NavigationBar) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Route
        component={[Function]}
        path="/add-recipe"
      />
      <Route
        component={[Function]}
        path="/user/:tab"
      />
      <Route
        component={[Function]}
        path="/update-recipe/:recipeId"
      />
      <Route
        component={[Function]}
        path="/user-recipes/:recipeId"
      />
      <Route
        component={[Function]}
        path="/recipe/:recipeId"
      />
      <Route
        component={[Function]}
        path="/all-recipes"
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
    <Footer />
  </div>
</BrowserRouter>
`;
